pepper-apis/src/main/java/org/broadinstitute/ddp/util/PubSubEmulator.java

```diff
@@ -149,6 +154,29 @@ public static void shutdown() throws IOException {
     }
 
     public static synchronized boolean hasStarted() {
-        return pubsubProcess != null;
+        // We might be running a Docker based emulator
+        if (isExternalEmulatorRunning()) {
+            LOG.info("An emulator already running at " + EMULATOR_HOST);
+            return true;
+        } else {
+            return pubsubProcess != null;
+        }
+    }
+
+    private static boolean isExternalEmulatorRunning() {
+        try (
+                CloseableHttpClient httpClient = HttpClients.createDefault();
+        ) {
+            CloseableHttpResponse response =
+                    httpClient.execute(new HttpGet("http://" + EMULATOR_HOST));
+            return response.getStatusLine().getStatusCode() == 200;
+        } catch (HttpHostConnectException e) {
+            LOG.debug("Could not connect to pubsub server. Must not be running");
+            return false;
+        } catch (IOException e) {
+            String msg = "There was problem initializing CloseableHttpClient";
+            LOG.error(msg, e);
+            throw new DDPException(msg, e);
+        }
```

Comment:
In CircleCi we are starting a pubsub emulator from a Docker image. If the port is being used we are going to assume a pubsub emulator already running and no need to start our own from Java.